File: Queue/codeforces_266B.py

# Problem link ===>> https://codeforces.com/problemset/problem/266/B

def queue_at_school(n, t, s):
  """Simulates the queue of students and returns the number of students who can be served in the allotted time.

  Args:
    n: The number of students in the queue.
    t: The allotted time.
    s: A string representing the initial arrangement of the students, where 'B' represents a boy and 'G' represents a girl.

  Returns:
    The number of students who can be served in the allotted time.
  """

  queue = []
  for i in range(n):
    if s[i] == 'B':
      queue.append(i)

  time = 0
  while queue and time <= t:
    current = queue.pop(0)
    time += 1

  return time


if __name__ == '__main__':
  n, t = map(int, input().split())
  s = input()

  result = queue_at_school(n, t, s)
  print(result)

File: Queue/codeforces_266B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/266/B
// submission link ===>> https://codeforces.com/contest/266/submission/241202383

// #include <iostream>
// #include <queue>
// using namespace std;

// int main()
// {
//     queue<int> queue;
//     int n, patience;

//     cin >> n;
//     for (int i = 0; i < n; i++)
//     {
//         cin >> patience;
//         queue.push(patience);
//     }

//     int left = 0;

//     while (!queue.empty())
//     {
//         int patience = queue.front();
//         queue.pop();

//         if (patience > 1)
//         {
//             patience--;
//             queue.push(patience);
//         }
//         else
//         {
//             left++;
//         }
//     }

//     cout << left << endl;

//     return 0;
// }

#include <iostream>

using namespace std;

int main()
{
    int n, t, i, j;
    cin >> n >> t;
    string s;

    for (i = 0; i < n; i++)
    {
        cin >> s;
    }

    for (i = 1; i <= t; i++)
    {

        for (j = 0; j < n - 1; j++)
        {

            if (s[j] == 'B' && s[j + 1] == 'G')
            {
                swap(s[j], s[j + 1]);
                j++;
            }
        }
    }
    cout << s << endl;

    return 0;
}